### PR TITLE
[intfsorch] Retrieve Port object before setting NAT zone on router in…

### PR DESCRIPTION
**Pull request by kperumalbfn**
Method(setRouterIntfsNatZoneId) is called using local Port object that causes orchagent to crash on config reload with non-zero NAT zone configuration.

Orchagent:
Retrieve Port object using getPort method before setting NAT zone on Router interfaces.

Addressing the issue : #1312
Signed-off-by: Kumaresh Perumal kumaresh.perumal@intel.com

### CODE REVIEW

**Comment by kperumalbfn** [comment]
retest this please

**Comment by arlakshm** [comment]
restest this please

File: orchagent/intfsorch.cpp [hunk 1]
```diff
@@ -712,21 +712,24 @@ void IntfsOrch::doTask(Consumer &consumer)
                     continue;
                 }
 
-                /* Set nat zone id */
-                if ((!nat_zone.empty()) and (port.m_nat_zone_id != nat_zone_id))
+                if (gPortsOrch->getPort(alias, port))
```
Comment (arlakshm) [inline orchagent/intfsorch.cpp:715]: this `gPortsOrch->getPort(alias, port))`check is already done above( line 665). Why do we need to do the check here again ?

Comment (kperumalbfn) [reply orchagent/intfsorch.cpp:715]: @arlakshm RIF objects are created in setIntf() before calling set_nat_zone API and the RIF object handle is set in port.m_rif_id. To get the Port object with rif handle, getPort() is invoked before setting the NAT zone.

**Comment by kperumalbfn** [comment]
Retest vs please

**Comment by kperumalbfn** [comment]
retest this please

**Comment by kperumalbfn** [comment]
retest this please

**Comment by kperumalbfn** [comment]
retest this please

**Comment by prsunny** [comment]
looks like this is required for 201911 as well, correct?

**Comment by arlakshm** [comment]
> looks like this is required for 201911 as well, correct?

@prsunny, yes this is required for 201911. I have added the label.

**Comment by kperumalbfn** [comment]
retest this please

**Comment by kperumalbfn** [comment]
retest this please

**Comment by kperumalbfn** [comment]
@arlakshm Could you merge this.

**Comment by abdosi** [comment]
@arlakshm Will wait for PR#1411 for 201911 first and then cherry-pick this.

**Comment by arlakshm** [comment]
> @arlakshm Will wait for PR#1411 for 201911 first and then cherry-pick this.

@abdosi, I think #1411 this PR are independent. We can cherry-pick this.

**Pull request merged**